docs/protocol/core/v3.0.rst

```diff
@@ -1,3 +1,152 @@
 Zarr core protocol version 3.0
 ==============================
 
+
+Conceptual model
+----------------
+
+A Zarr *hierarchy* is a tree structure, where each node in the tree is
+either a *group* or an *array*. Group nodes may have children
+but array nodes may not.
+
+Each node in a hierarchy has a *name* which is a string of ASCII
+characters with some additional constraints. Two sibling nodes cannot 
+have the same name. The root node does not have a name.
+
+Each node in a hierarchy has a *path* which uniquely identifies that
+node and defines its location within the hierarchy. The path is formed 
+by joining together the "/" character, followed by the names of all 
+ancestor nodes separated by the "/" character, followed by the name of 
+the node itself. For example, the path "/foo/bar" identifies a node 
+named "bar", whose parent is named "foo", whose parent is the root of 
+the hierarchy. The string "/" identifies the root node.
+
+An array has a fixed number of *dimensions*. Each dimension has an
```

Comment:
Is this fixed number >0, i.e. are zero-rank arrays allowed?

Reply:
I would say zero-rank arrays should be allowed in the core protocol. I.e., change this sentence to:

```suggestion
An array has a fixed number of zero or more *dimensions*. Each dimension has an
```

However, happy to discuss either way.

Reply:
Is there a good reason to support rank zero?
I can't think of a use-case right now.

Reply:
I believe it's used for scalar values. We added support for this in the python implementation - https://github.com/zarr-developers/zarr/pull/161. 

(Note there was some mixing of discussion there as we also added support for arrays with one or more zero-length dimensions at around the same time and the two requirements got a bit confused, see also - https://github.com/zarr-developers/zarr/pull/160.)

Reply:
That makes sense. Thanks for the clarification.